src/meta/src/stream/stream_manager.rs

```diff
@@ -483,13 +483,19 @@ impl GlobalStreamManager {
 
         let dummy_table_id = table_fragments.table_id();
 
+        let init_split_assignment = self
```

Comment:
What's the purpose of these code? Would you mind adding some comments here?

Reply:
Yes we need to do this since for source executors, the split info is only fetched via barriers. And for `Replace`, the barrier is the `Update` mutation from `ReplaceTable`, which means it needs the split info to make source executors start to read. `pre_allocate_splits` does two things: fetch latest discovered splits and allocate. 

Splits are not allocated until this line IIUC.

I think this line is just like the [code](https://github.com/risingwavelabs/risingwave/blob/3c0284ee4c3575f2b00f7d0ae0d9fa648e844858/src/meta/src/stream/stream_manager.rs#L442) for creating table.

Reply:
since we change the fragment and it seems to be a new entry for source info, we need to call this func to persist meta data into etcd.